habitat/tasks/nav/multi_object_nav_task.py

```diff
@@ -0,0 +1,551 @@
+            [
+                self._dataset.category_to_task_category_id[i]
+                for i in category_name
+            ],
+            dtype=np.int64,
+        )
+        return goals[task.current_goal_index : task.current_goal_index + 1]
+
+
+@registry.register_measure
+class DistanceToCurrentObjectGoal(Measure):
+    """Calculates distance from the current object goal."""
+
+    cls_uuid: str = "distance_to_current_object_goal"
+
+    def __init__(
+        self, sim: Simulator, config: Config, *args: Any, **kwargs: Any
+    ):
+        self._sim = sim
+        self._config = config
+        super().__init__(**kwargs)
+
+    def _get_uuid(self, *args: Any, **kwargs: Any):
+        return self.cls_uuid
+
+    def reset_metric(
+        self,
+        episode,
+        task,
+        *args: Any,
+        **kwargs: Any,
+    ):
+        self._metric = None
+        self.update_metric(*args, episode=episode, task=task, **kwargs)
+
+    def _euclidean_distance(self, position_a, position_b):
+        return np.linalg.norm(
+            np.array(position_b) - np.array(position_a), ord=2
+        )
+
+    def update_metric(
+        self,
+        *args: Any,
+        episode: MultiObjectGoalNavEpisode,
+        task: MultiObjectNavigationTask,
+        **kwargs: Any,
+    ):
+        current_position = self._sim.get_agent_state().position.tolist()
+        distance_to_current_goal = self._sim.geodesic_distance(
+            current_position, episode.goals[task.current_goal_index].position
+        )
+
+        self._metric = distance_to_current_goal
+
+
+@registry.register_measure
+class CurrentGoalSuccess(Measure):
+    r"""Whether or not the agent succeeded in finding it's
+    current goal. This measure depends on DistanceToCurrentObjectGoal measure.
+    """
+
+    cls_uuid: str = "current_goal_success"
+
+    def __init__(
+        self, *args: Any, sim: Simulator, config: Config, **kwargs: Any
+    ):
+        self._sim = sim
+        self._config = config
+
+        super().__init__()
+
+    def _get_uuid(self, *args: Any, **kwargs: Any):
+        return self.cls_uuid
+
+    def reset_metric(self, *args: Any, episode, task, **kwargs: Any):
+        task.measurements.check_measure_dependencies(
+            self.uuid, [DistanceToCurrentObjectGoal.cls_uuid]
+        )
+        self.update_metric(*args, episode=episode, task=task, **kwargs)
+
+    def update_metric(
+        self,
+        *args: Any,
+        episode,
+        task: MultiObjectNavigationTask,
+        **kwargs: Any,
+    ):
+        distance_to_current_goal = task.measurements.measures[
+            DistanceToCurrentObjectGoal.cls_uuid
+        ].get_metric()
+
+        if (
+            task.is_found_called
+            and distance_to_current_goal <= self._config.SUCCESS_DISTANCE
+        ):
+            self._metric = 1
+            task.current_goal_index += 1
+        else:
+            self._metric = 0
+
+
+@registry.register_measure
+class Progress(Measure):
```

Comment:
Either give this a MultiON specific name or generalize it to other navigation tasks.